pilot/pkg/networking/plugin/authz/rbac.go

```diff
@@ -199,18 +197,53 @@ func convertRbacRulesToFilterConfig(service string, roles []model.Config, bindin
 				permissions = append(permissions, convertToPermission(rule))
 			}
 		}
+		if len(permissions) == 0 {
+			log.Debugf("role %v skipped for no permissions found", role.Name)
+			continue
+		}
 
-		if len(permissions) != 0 {
+		enforcedPrincipals, permissivePrincipals := convertToPrincipals(roleToBinding[role.Name])
+		principals := make([]*policyproto.Principal, 0)
+		principals = append(principals, enforcedPrincipals...)
+		principals = append(principals, permissivePrincipals...)
+
+		if len(principals) == 0 {
```

Comment:
Why move this from line 186 to 210? If len(principals) == 0 we could return early and no need to call `convertToPermission()`.

Reply:
call convertToPrincipals or convertToPermission first looks the same to me(neither of them is heavy)..

Reply:
The stringMatch in line 194 and the TODO to add more matching behavior makes it more expensive to check the permissions first.

Reply:
maybe, not worth arguing on this anyway :) updated.